src/Miunie.Core/Services/LanguageResources.cs

```diff
@@ -9,37 +10,43 @@ public class LanguageResources : ILanguageResources
 
         private Random _rand;
         private readonly string _collection = "Lang";
+        private readonly string _langKeyFormat = "Phrases{0}";
+        private string _langKey = "PhrasesEn";
 
         public LanguageResources(IDataStorage storage, Random rand)
         {
             _storage = storage;
             _rand = rand;
         }
+        public void SetLanguage(string langKey)
+            => _langKey = GetLangKey(langKey);
 
-        public string GetPhrase(string key)
+        public string GetPhrase(string key, params object[] objs)
         {
-            var phrases =_storage.RestoreObject<string[]>(_collection, key);
-            return PickRandom(phrases);
+            var phrases = GetFromStorage(key);
+            var phrase = PickRandom(phrases);
+            return String.Format(phrase, objs);
         }
 
-        private string PickRandom(string[] collection)
+        private string[] GetFromStorage(string key)
         {
-            if(collection is null)
-            {
-                return String.Empty;
-            }
-            if(collection.Length == 1)
-            {
-                return collection[0];
-            }
-            var index = _rand.Next(collection.Length);
-            return collection[index];
+            var phrases = _storage.RestoreObject<Dictionary<string, string[]>>(
+                                                                    _collection,
+                                                                    _langKey);
+
+            phrases.TryGetValue(key, out var values);
+            return values;
         }
 
-        public string GetFormatted(string key, params object[] objs)
+        private string GetLangKey(string langKey)
+            => String.Format(_langKeyFormat, langKey);
+
+        private string PickRandom(string[] collection)
         {
-            var phrase = GetPhrase(key);
-            return String.Format(phrase, objs);
+            if (collection is null) { return String.Empty; }
+            if (collection.Length == 1) { return collection[0]; }
```

Comment:
This seems redudent as `_rand.Next(collection.Length);` will always return the first and only index if only 1 item is in the collection.